x-pack/solutions/search/plugins/search_synonyms/public/hooks/use_put_synonyms_set.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { SynonymsPutSynonymResponse } from '@elastic/elasticsearch/lib/api/types';
+import { i18n } from '@kbn/i18n';
+import { KibanaServerError } from '@kbn/kibana-utils-plugin/common';
+import { PLUGIN_ROUTE_ROOT } from '../../common/api_routes';
+import { useKibana } from './use_kibana';
+
+interface MutationArgs {
+  synonymsSetId: string;
+}
+
+export const usePutSynonymsSet = (onSuccess?: () => void, onError?: (error: string) => void) => {
+  const queryClient = useQueryClient();
+  const {
+    services: { http, notifications, application },
+  } = useKibana();
+
+  return useMutation(
+    async ({ synonymsSetId }: MutationArgs) => {
+      return await http.put<SynonymsPutSynonymResponse>(
+        `/internal/search_synonyms/synonyms/${synonymsSetId}`
+      );
+    },
+    {
+      onSuccess: (_, { synonymsSetId }) => {
+        queryClient.invalidateQueries(['synonyms-sets-fetch']);
+        queryClient.invalidateQueries(['synonyms-rule-fetch']);
+        notifications?.toasts?.addSuccess({
+          title: i18n.translate('xpack.searchSynonyms.putSynonymsSetSuccess', {
+            defaultMessage: 'Synonyms set {synonymsSetId} put',
+            values: { synonymsSetId },
+          }),
+        });
+        if (onSuccess) {
+          onSuccess();
+        }
+        application.navigateToUrl(`${PLUGIN_ROUTE_ROOT}/sets/${synonymsSetId}`);
```

Comment:
:x: these need `http.basePath.prepend` to navigate to the URL in hosted with or without a space defined.